File: src/C#/BehavioralPatterns/Mediator/CQRS/Mediator.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CQRS
{
    public class Mediator : IMediator
    {
        private readonly Dictionary<Type, Type> requestHandlers;

        public Mediator(Dictionary<Type, Type> requestHandlers)
        {
            this.requestHandlers = requestHandlers;
        }

        public object Send(IRequest request)
        {
            var requestHandlerType = requestHandlers[request.GetType()];
            var requestHandler = Activator.CreateInstance(requestHandlerType);
            return requestHandlerType.GetMethod("Execute").Invoke(requestHandler, new object[] { request });
        }
    }
}

File: src/JavaScript/BehavioralPatterns/Mediator/CQRS/mediator.js

class Mediator {
    constructor(requestHandlers) {
        this.requestHandlers = requestHandlers;
    }

    send(request) {
        const requestHandlerType = this.requestHandlers[request.constructor.name];

        if (!requestHandlerType) {
            throw new Error("Handler not found for request type.");
        }
        const requestHandler = new requestHandlerType();
        return requestHandler.execute(request);
    }
}

module.exports = Mediator;